packages/gatsby-theme-docs/src/components/link.spec.js

```diff
@@ -0,0 +1,134 @@
+  beforeEach(() => {
+    useSiteData.mockClear();
+    useSiteData.mockReturnValue({ pathPrefix });
+  });
+  const scenarios = [
+    {
+      title: 'image link',
+      props: { href: '/static/123' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'image-link' },
+    },
+    {
+      title: 'external link',
+      props: {
+        href: 'https://github.com/commercetools/commercetools-docs-kit',
+      },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'external-link' },
+    },
+    {
+      title: 'anchor link',
+      props: { href: '#title-1' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'anchor-link' },
+    },
+    {
+      title: 'anchor link to same page',
+      props: { href: '/page-1/' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'anchor-link', href: '' },
+    },
+    {
+      title: 'anchor link empty',
+      props: { href: '' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'anchor-link' },
+    },
+    {
+      title: 'history link to another page',
+      props: { href: '/page-2' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'history-link' },
+    },
+    {
+      title: 'history link to a sub-page',
+      props: { href: 'getting-started' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'history-link', href: '/page-1/getting-started' },
+    },
+    {
+      title: 'history link to a sub-page using a trailing slash',
+      props: { href: 'getting-started/' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'history-link', href: '/page-1/getting-started/' },
+    },
+    {
+      title: 'history link to a parent page',
+      props: { href: '../page-2' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'history-link', href: '/page-2' },
+    },
+    {
+      title: 'history link to another page with hash',
+      props: { href: '/page-2#title' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'history-link', href: '/page-2#title' },
+    },
+    {
+      title: 'internal link',
+      props: { href: 'https://docs.commercetools.com/site-key/page-1' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'internal-link', href: '/site-key/page-1' },
+    },
+    {
+      title: 'internal link using "/../" notation',
+      props: { href: '/../site-key/page-1' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'internal-link', href: '/site-key/page-1' },
+    },
+    {
+      title: 'internal link with hash',
+      props: { href: 'https://docs.commercetools.com/site-key/page-1#title' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'internal-link', href: '/site-key/page-1#title' },
+    },
+  ];
+
+  const scenariosInProdMode = scenarios.map(scenario => ({
+    ...scenario,
+    location: { pathname: withPrefix(scenario.location.pathname) },
```

Comment:
if you keep the assertion generation magic I think we need to cover the special case we spoke about (a link to a different microsite would be kept as an absolute URL in dev mode but we would remove the host for production because otherwise we can't test refactors that span across multiple sites)

Reply:
Ah right, I forgot about the dev/prod mode difference